### PR TITLE
Simplify and Optimize RestController Slightly

### PR DESCRIPTION
**Pull request by original-brownbear**
* Simplify the path iterator to generate less garbage
* `dispatchRequest` always terminates, adjust code accordingly

### CODE REVIEW

**Comment by original-brownbear** [comment]
Jenkins run elasticsearch-ci/2

**Comment by original-brownbear** [comment]
Jenkins run elasticsearch-ci/2

File: server/src/main/java/org/elasticsearch/rest/RestController.java [hunk 1]
```diff
@@ -199,18 +197,18 @@ public void dispatchBadRequest(final RestChannel channel, final ThreadContext th
     /**
      * Dispatch the request, if possible, returning true if a response was sent or false otherwise.
```
Comment (dakrone) [inline server/src/main/java/org/elasticsearch/rest/RestController.java]: This comment is incorrect now about returning a boolean

**Review by dakrone — APPROVED** [review]
LGTM, left one comment about the javadoc

**Comment by original-brownbear** [comment]
thanks @dakrone !

**Pull request merged**